src/modules/70-pipeline/components/ArtifactsSelection/ArtifactUtils.ts

```diff
@@ -129,46 +130,93 @@ export const getFinalArtifactObj = (
   return artifactObj
 }
 
-export const getFinalArtifactFormObj = (
-  formData: ImagePathTypes & { connectorId?: string },
-  isSideCar: boolean
-): ArtifactConfig => {
-  const tagData =
+const getServerlessArtifactFromObj = (formData: ImagePathTypes & { connectorId?: string }): ArtifactConfig => {
+  const artifactPathData =
     formData?.tagType === TagTypes.Value
-      ? { tag: defaultTo(formData.tag?.value, formData.tag) }
-      : { tagRegex: defaultTo(formData.tagRegex?.value, formData.tagRegex) }
+      ? { artifactPath: defaultTo(formData.tag?.value, formData.tag) }
+      : {
+          artifactPathFilter: defaultTo(formData.tagRegex?.value, formData.tagRegex)
+        }
 
-  const artifactObj: ArtifactConfig = {
+  return {
     spec: {
       connectorRef: formData?.connectorId,
-      artifactPath: formData?.artifactPath,
-      ...tagData
+      artifactDirectory: formData?.artifactDirectory,
+      ...artifactPathData
+    }
+  }
+}
+
+export const getFinalArtifactFormObj = (
+  formData: ImagePathTypes & { connectorId?: string },
+  isSideCar: boolean,
+  isServerlessDeploymentTypeSelected = false
+): ArtifactConfig => {
+  let artifactObj: ArtifactConfig = {}
+
+  if (isServerlessDeploymentTypeSelected) {
+    artifactObj = getServerlessArtifactFromObj(formData)
+  } else {
+    const tagData =
+      formData?.tagType === TagTypes.Value
+        ? { tag: defaultTo(formData.tag?.value, formData.tag) }
+        : { tagRegex: defaultTo(formData.tagRegex?.value, formData.tagRegex) }
+
+    artifactObj = {
+      spec: {
+        connectorRef: formData?.connectorId,
+        artifactPath: formData?.artifactPath,
+        ...tagData
+      }
     }
   }
+
   if (isSideCar) {
     merge(artifactObj, { identifier: formData?.identifier })
   }
   return artifactObj
 }
 
+const getTagValues = (specValues: any, isServerlessDeploymentTypeSelected = false): ImagePathTypes => {
```

Comment:
Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232297)
```suggestion
const getTagValues = (specValues: unknown, isServerlessDeploymentTypeSelected = false): ImagePathTypes => {
```